### PR TITLE
bf: ZENKO-903 retrieve new bucket list on report

### PR DESCRIPTION
**Pull request by alexanderchan-scality**

### CODE REVIEW

**Review by anurag4DSB — APPROVED** [review]

**Review by anurag4DSB — APPROVED** [review]

**Comment by alexanderchan-scality** [comment]
@bert-e  wait

**Review by JianqinWang — APPROVED** [review]

File: lib/storage/metadata/mongoclient/MongoClientInterface.js [hunk 1]
```diff
@@ -1205,10 +1212,17 @@ class MongoClientInterface {
                 if (err) {
                     return cb(err);
                 }
+
+                if (!doFullScan) {
+                    const cachedRes = this.dataCount.results();
+                    cachedRes.bucketList = [...res.bucketList];
+                    cachedRes.buckest = res.buckets;
```
Comment (rahulreddy) [inline lib/storage/metadata/mongoclient/MongoClientInterface.js]: `buckets` is misspelt as `buckest`

**Review by rahulreddy — CHANGES_REQUESTED** [review]

File: lib/storage/metadata/mongoclient/MongoClientInterface.js [hunk 2]
```diff
@@ -1205,10 +1212,17 @@ class MongoClientInterface {
                 if (err) {
                     return cb(err);
                 }
+
+                if (!doFullScan) {
+                    const cachedRes = this.dataCount.results();
+                    cachedRes.bucketList = [...res.bucketList];
```
Comment (bennettbuchanan) [inline lib/storage/metadata/mongoclient/MongoClientInterface.js]: Minor: Is the spread operator necessary?

**Review by bennettbuchanan — APPROVED** [review]

File: lib/storage/metadata/mongoclient/MongoClientInterface.js (hunk 2 above)
Comment (alexanderchan-scality) [reply lib/storage/metadata/mongoclient/MongoClientInterface.js]: Looking at the code, at that point, there won't be any modification of the `res` object, so the creation of a new object with spread is not necessary. Will fix.

**Review by rahulreddy — APPROVED** [review]

**Pull request merged**